core/commands/repository/interactors/erase_repository.py

```diff
@@ -0,0 +1,33 @@
+from django.conf import settings
+
+import services.self_hosted as self_hosted
+from api.shared.permissions import UserIsAdminPermissions
+from codecov.commands.base import BaseInteractor
+from codecov.commands.exceptions import Unauthenticated, Unauthorized, ValidationError
+from codecov.db import sync_to_async
+from codecov_auth.helpers import current_user_part_of_org
+from codecov_auth.models import Owner
+from core.models import Repository
+from services.task.task import TaskService
+
+
+class EraseRepositoryInteractor(BaseInteractor):
+    def validate_owner(self, owner: Owner):
+        if not current_user_part_of_org(self.current_owner, owner):
```

Comment:
I noticed for a lot of Interactors we also add code like this:

https://github.com/codecov/codecov-api/blob/a29cd709908af81dac427a30dff8c6436480373f/codecov_auth/commands/owner/interactors/update_default_organization.py#L15-L16

Is that not needed here?

Reply:
I had it initially and realized that the `@require_authenticated` decorator takes care of that. 

The other code is a result of what I had originally thought so I will remove that.